packages/tradershub/src/hooks/useMT5AccountHandler.ts

```diff
@@ -0,0 +1,92 @@
+import {
+    useAccountStatus,
+    useActiveTradingAccount,
+    useAvailableMT5Accounts,
+    useCreateMT5Account,
+    useSettings,
+    useTradingPlatformPasswordChange,
+} from '@deriv/api';
+import { MarketType, PlatformDetails } from '../features/cfd/constants';
+import { Jurisdiction } from '../features/cfd/screens/CFDCompareAccounts/constants';
+import { TMarketTypes } from '../types';
+
+type TUseMT5AccountHandler = {
+    marketType: TMarketTypes.All;
+    selectedJurisdiction: string;
+};
+
+const useMT5AccountHandler = ({ marketType, selectedJurisdiction }: TUseMT5AccountHandler) => {
+    const {
+        data: accountStatus,
+        error: isCreateMT5AccountError,
+        isLoading: createMT5AccountLoading,
+        isSuccess: isCreateMT5AccountSuccess,
+        status,
+    } = useAccountStatus();
+    const { mutate: createMT5Account } = useCreateMT5Account();
+    const { isLoading: tradingPlatformPasswordChangeLoading, mutateAsync: tradingPasswordChange } =
+        useTradingPlatformPasswordChange();
+    const { data: activeTrading } = useActiveTradingAccount();
+    const { data: settings } = useSettings();
+    const { data: availableMT5Accounts } = useAvailableMT5Accounts();
+    const isMT5PasswordNotSet = accountStatus?.is_mt5_password_not_set;
+
+    const accountType = marketType === MarketType.SYNTHETIC ? 'gaming' : marketType;
+    const categoryAccountType = activeTrading?.is_virtual ? 'demo' : accountType;
+
+    const handleSubmit = (password: string) => {
+        // in order to create account, we need to set a password through trading_platform_password_change endpoint first
+        // then only mt5_create_account can be called, otherwise it will response an error for password required
+        if (isMT5PasswordNotSet) {
+            return tradingPasswordChange({
+                new_password: password,
+                platform: PlatformDetails.mt5.platform,
```

Comment:
Why are we using 2 different constant for one value.
In some places we use `CFDPlatforms.MT5` and here we are using `PlatformDetails.mt5.platform`
is there any specific reason for this

Reply:
this should be coming from `CFDPlatforms.MT5` as we only need the value 'mt5' and not the other details coming from `PlatformDetails.mt5`